### PR TITLE
Add event to notify that the `switch` has been enabled

### LINKED ISSUE #152: Enabling the interfaces but not enabling the switch makes flow_manager not install of_lldp flows
Dear guys,

The topology offers an API to enable all the interface of a switch: https://github.com/kytos/topology/blob/master/openapi.yml#L239-L258

From the documentation it is not clear that this API call should also enable the switch, but it does enable the switch indeed (not sure if this is a bug or by design). However, when following this step, the flows for of_lldp are not actually enabled on the switch (sounds related to https://github.com/kytos/of_lldp/issues/60)

Steps to reproduce:

1. Start Kytos (without -E):
```
docker run -d -p 6653:6653 -p 8181:8181 --name k1 amlight/kytos:latest
```
2. Start Mininet (any topology):
```
sudo mn --topo linear,4 --controller=remote,ip=127.0.0.1
```
3. Enable all the interfaces for each switch:
```
for i in $(curl -s http://127.0.0.1:8181/api/kytos/topology/v3/switches | jq -r '.switches[].id'); do curl -X POST -H 'Content-Type: application/json' http://127.0.0.1:8181/api/kytos/topology/v3/interfaces/switch/$i/enable; done
```

Expected behavior:

Not sure if those steps should leave the switch disable (since the API call was supposed to enable the interfaces not the switch itself) or enable the switch indeed - raising the proper events and allowing the of_lldp to install its flows.

Actual behavior:

1. All the switches are enabled:
```
$ curl -s http://127.0.0.1:8181/api/kytos/topology/v3/switches | jq -r '.switches[].enabled'
true
true
true
true
```

2. However, the of_lldp flows are not created:
```
mininet-wifi> sh ovs-ofctl dump-flows s1
mininet-wifi> sh ovs-ofctl dump-flows s2
mininet-wifi> sh ovs-ofctl dump-flows s3
mininet-wifi> sh ovs-ofctl dump-flows s4
mininet-wifi>
```

Workaround: if you follow the "normal" workflow of 1st enabling the switch and then enabling its interfaces, the of_lldp flows are properly setup:
```
for i in $(curl -s http://127.0.0.1:8181/api/kytos/topology/v3/switches | jq -r '.switches[].id'); do curl -X POST -H 'Content-Type: application/json' http://127.0.0.1:8181/api/kytos/topology/v3/switches/$i/enable; curl -X POST -H 'Content-Type: application/json' http://127.0.0.1:8181/api/kytos/topology/v3/interfaces/switch/$i/enable; done
```

### PR DESCRIPTION
**Pull request by cmagnobarbosa**
# Pull Request Template

### :octocat: Are you working on some issue? Identify the issue!

Fix #152 

### :bookmark_tabs: Description of the Change

Add an event to notify that enabling an `interface` also enables the `switch`.

### :computer: Verification Process

- Unit tests performed.

### :page_facing_up: Release Notes

- Add an event to notify that enabling an `interface` also enables the ` switch`.

### CODE REVIEW

**Review by josemauro — APPROVED** [review]

File: main.py [hunk 1]
```diff
@@ -340,6 +340,7 @@ def enable_interface(self, interface_enable_id=None, dpid=None):
         if not error_list:
             log.info(f"Storing administrative state for enabled interfaces.")
             self.save_status_on_storehouse()
+            self.notify_switch_enabled(dpid)  # The switch is also enabled
```
Comment (hdiogenes) [inline main.py:343]: Should an interface_enable really trigger a switch_enabled  event?

Comment (cmagnobarbosa) [reply main.py:343]: Today, when an interface connected to a disabled switch is enabled, the switch is also enabled, but this activation does not trigger the `switch_enabled` event. Therefore, it is necessary to trigger this event here.

Comment (hdiogenes) [reply main.py:343]: After talking with @beraldoleal we decided that this behavior is wrong. The right thing to do is to raise an error if the user tries to enable an interface in a disabled switch.

Comment (cmagnobarbosa) [reply main.py:343]: An issue was created to change this behavior https://github.com/kytos/kytos/issues/1233

**Pull request closed without merging**